### PR TITLE
Update checkout_account.py to not reuse connections

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:
Currently the automated e2e tests are having issues related reusing the httplib connection: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-cluster-api-provider-aws-bazel-e2e/1113802682641944577, this udpates checkout_account.py to not reuse connections.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: hack/checkout_account.py [hunk 1]
```diff
@@ -36,22 +37,21 @@ def post_request(conn, input_state = "free"):
     })
 
     )
-    return conn.getresponse()
+    resp = conn.getresponse()
+    conn.close()
```
Comment (ncdc) [inline hack/checkout_account.py]: I'm fairly certain that if you close the connection before reading the response (done below in `json.load(resp)`), that you won't get any of the data.

Comment (detiber) [reply hack/checkout_account.py]: arrgh, of course it would behave that way :joy: Let me fix up to return the json.

**Comment by detiber** [comment]
@ncdc it isn't pretty, but it should work

**Comment by ncdc** [comment]
/lgtm
fingers crossed

**Pull request merged**